src/irmin-http/swagger.yml

```diff
@@ -17,90 +17,193 @@ paths:
+          description: "branch already exists, no-op"
         201:
-          description: "" # TODO
+          description: "success"
   /branch/{branchName}:
+    parameters:
+    - in: "path"
+      name: "branchName"
+      description: "name of branch to operate on"
+      required: true
+      type: "string"
     get:
       summary: "get a branch by name"
+      consumes:
+      - "application/json"
+      produces:
+      - "application/json"
       responses:
         200:
-          description: "" #TODO
+          description: "success"
+          schema:
+            $ref: "#/definitions/Branch"
         404:
-          description: "" #TODO
+          description: "branch not found"
     delete:
       summary: "delete a branch by its ID"
+      consumes:
+      - "application/json"
+      produces:
+      - "application/json"
       responses:
         200:
-          description: "" #TODO
+          description: "success"
         404:
-          description: "" #TODO
+          description: "branch not found"
   /commits:
     get:
-      summary: "lists all commits"
+      summary: "lists all commits, newest to oldest (like `git log`)"
+      consumes:
+      - "application/json"
+      produces:
+      - "application/json"
       responses:
         200:
-          description: "" #TODO
-        404:
-          description: "" #TODO
+          description: "success"
+          schema:
+            type: array
+            items:
+              $ref: "#/definitions/Commit"
   /commits/{commitHash}:
+    parameters:
+    - in: "path"
+      name: "commitHash"
+      description: "hash of commit to operate on"
+      required: true
+      type: "string"
     get:
       summary: "get a commit by its hash"
+      consumes:
+      - "application/json"
+      produces:
+      - "application/json"
       responses:
         200:
-          description: "" #TODO
+          description: "success"
+          schema:
+            $ref: "#/definitions/Commit"
         404:
-          description: "" #TODO
-  /paths:
+          description: "commit not found"
+  /keys:
     get:
       summary: "gets all keys and their associated values (like `irmin tree`)"
+      consumes:
+      - "application/json"
+      produces:
+      - "application/json"
       responses:
         200:
-          description: "" #TODO
-        404:
-          description: "" #TODO
+          description: "success"
+          schema:
+            type: array
+            items:
+              $ref: "#/definitions/KV"
     post:
       summary: "add a new key/value pair to the store"
+      consumes:
+      - "application/json"
+      produces:
+      - "application/json"
+      parameters:
+      - in: body
+        name: body
+        description: "key/value pair to add"
+        required: true
+        schema:
+          $ref: "#/definitions/KV"
       responses:
         200:
-          description: "" #TODO
+          description: "key already exists, no-op"
```

Comment:
In my point of view, I consider Irmin as a `key -> value` store like a functionnal [Map](https://caml.inria.fr/pub/docs/manual-ocaml/libref/Map.Make.html) available from the standard library (but much more complex indeed). In this view, when you want to add a value already bound in this Irmin database, nothing happen but it's not error. We can reduce this operation as an `identity` function.

So, the _state_ in the client side and in the server side would not change. But, as I said, we need to keep the validity of this operation (if we follow `Map.add`). And it's why I consider the 204 HTTP code, like it's not an error but nothing happen in both side.

Then, I'm not a killer API conceptor 😄 !